### PR TITLE
feat: Allow injection of a different base host url and custom headers

### PR DESCRIPTION
**Pull request by fnazarios**
This PR changes the code to allow us pass a different Open AI api host and also to inject custom headers.

### CODE REVIEW

File: Sources/OpenAIKit/Configuration.swift [hunk 1]
```diff
@@ -1,26 +1,33 @@
 import NIOHTTP1
+import Foundation
 
 public struct Configuration {
     public let apiKey: String
     public let organization: String?
+    public let host: String
+    var headers: HTTPHeaders
```
Comment (cs-workco) [inline Sources/OpenAIKit/Configuration.swift]: this can be a let OR you don't need the local `headers` property in the init. Right?

Comment (fnazarios) [reply Sources/OpenAIKit/Configuration.swift]: We do need `headers` here. I already updated it to let

**Review by cs-workco — APPROVED** [review]
👍🏽

**Review by leoreisbk — APPROVED** [review]
LGTM 🚀

**Pull request merged**